src/Validators/IntegrityChecks/Form/OptionalIfCheck.cs

```diff
@@ -0,0 +1,81 @@
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using form_builder.Models;
+using form_builder.Models.Elements;
+
+namespace form_builder.Validators.IntegrityChecks.Form
+{
+    public class OptionalIfCheck : IFormSchemaIntegrityCheck
+    {
+        public IntegrityCheckResult Validate(FormSchema schema)
+        {
+            IntegrityCheckResult result = new();
+            List<string> questionIds = new();
+
+            List<IElement> optionalIfProperties = schema.Pages
```

Comment:
The names of these 3 lists are misleading, they are lists of elements, not lists of properties